ts/features/bonus/cgn/saga/activation/activationSaga.ts

```diff
@@ -1,20 +1,34 @@
 import { SagaIterator } from "redux-saga";
-import { call, put, race, take } from "redux-saga/effects";
+import { call, delay, put, race, take } from "redux-saga/effects";
 import { NavigationActions } from "react-navigation";
 import { navigationHistoryPop } from "../../../../../store/actions/navigationHistory";
 import {
   cgnActivationCancel,
-  cgnActivationStatus
+  cgnRequestActivation
 } from "../../store/actions/activation";
-import { navigateToCgnOnboardingInformationTos } from "../../navigation/actions";
+import {
+  navigateToCgnActivationCompleted,
+  navigateToCgnActivationInformationTos,
+  navigateToCgnActivationLoading
+} from "../../navigation/actions";
 
 export function* cgnStartActivationWorker() {
-  yield put(navigateToCgnOnboardingInformationTos());
+  yield put(navigateToCgnActivationInformationTos());
+  yield put(navigationHistoryPop(1));
+
+  yield take(cgnRequestActivation.request);
+
+  yield put(navigateToCgnActivationLoading());
   yield put(navigationHistoryPop(1));
 
-  // TODO :HERE WILL BE IMPLEMENTED THE ACTIVATION LOGIC
-  // WIP remove this instruction when the activation flow will be implemented
-  yield take(cgnActivationStatus.request);
+  // UNCOMMENT when api implementation has been completed
```

Comment:
Since more IDEs parse TODO comments, using TODO to mark future changes it's a good pattern

![Schermata 2021-02-02 alle 09 19 14](https://user-images.githubusercontent.com/822471/106571871-fdc3b400-6537-11eb-9a8b-19eb347ff091.png)

Reply:
Saga received a massive refactoring when it has been completely implemented the activation workflow in #2749

I understand the point about TODO/FIXME anyway! I'm going to be more accurate in future